packages/react-components/src/Grid.tsx

```diff
@@ -19,10 +30,30 @@ function Grid<TItem = GridDefaultItem>(
   props: GridProps<TItem>,
   ref: ForwardedRef<GridElement<TItem>>,
 ): ReactElement | null {
-  const [portals, rowDetailsRenderer] = useModelRenderer(props.rowDetailsRenderer, { renderSync: true });
+  const [portals, rowDetailsRenderer] = useModelRenderer(props.rowDetailsRenderer);
+
+  const innerRef = useRef<GridElement & { _recalculateColumnWidths?(...args: any[]): void }>(null);
+  const finalRef = useMergedRefs(innerRef, ref);
+
+  const [pendingRecalculateColumnWidthsCall, setPendingRecalculateColumnWidthsCall] = useState<any[] | null>(null);
+
+  useLayoutEffect(() => {
```

Comment:
This change introduces a bit of jumpiness to Grid since the column auto-widths are calculated asynchronously.

https://github.com/user-attachments/assets/eca42b54-d233-40fb-9459-4597715e1a96